lte/gateway/c/session_manager/LocalEnforcer.cpp

```diff
@@ -1266,10 +1279,88 @@ void LocalEnforcer::terminate_multiple_sessions(
   }
 }
 
+void LocalEnforcer::remove_rules_for_multiple_suspended_credit(
+    SessionMap& session_map,
+    std::unordered_set<
+        ImsiSessionIDAndCreditkey, ImsiSessionIDAndCreditkey_hash>&
+        suspended_credits,
+    SessionUpdate& session_update) {
+  for (const auto& suspended_credit : suspended_credits) {
+    const auto& imsi       = suspended_credit.imsi;
+    const auto& session_id = suspended_credit.session_id;
+    const CreditKey& ckey  = suspended_credit.cKey;
+
+    SessionSearchCriteria criteria(imsi, IMSI_AND_SESSION_ID, session_id);
+    auto session_it = session_store_.find_session(session_map, criteria);
+    if (!session_it) {
+      MLOG(MWARNING) << "Session " << session_id
+                     << " not found for termination";
+      return;
+    }
+    auto& session    = **session_it;
+    auto& session_uc = session_update[imsi][session_id];
+    remove_rules_for_suspended_credit(session, ckey, session_uc);
+  }
+}
+
+void LocalEnforcer::remove_rules_for_suspended_credit(
+    const std::unique_ptr<SessionState>& session, const CreditKey& ckey,
+    SessionStateUpdateCriteria& session_uc) {
+  // suspend this specific credit
+  session->set_suspend_credit(ckey, true, session_uc);
+
+  // check if this is the last credit. That will trigger Final Unit Actions
+  session->suspend_service_if_necesary_for_credit(ckey, session_uc);
```

Comment:
Do we not care about the return value here?

Reply:
no, the result will be taken care by the next iteration